src/bluesky/run_engine.py

```diff
@@ -2338,6 +2343,13 @@ async def _wait(self, msg):
         else:
             group = msg.kwargs["group"]
             move_on = msg.kwargs.get("move_on", False)
+
+        if isinstance(group, list):
```

Comment:
The codebase is generally tolerant of different types of collections. For example, if someone accumulated status objects in a `set` or a `tuple` I think there is no reason to reject that. Perhaps this could check for `isinstance(group, collections.abc.Iterable) and not isinstance(group, str)`?

Reply:
@danielballan That makes sense. I changed this to -- if isinstance(group, collections.abc.Iterable): and pushed.